Client/YooChooseNotifier.php

```diff
@@ -175,4 +177,62 @@ private function getNotificationEndpoint()
             $this->options['customer-id']
         );
     }
+
+    /**
+     * Returns the YooChoose recommendation endpoint
+     *
+     * @return string
+     */
+    private function getRecommendationEndpoint()
+    {
+        return sprintf(
+           '%s/api/%s',
+            rtrim( $this->options['reco-api-endpoint'], '/' ),
+            $this->options['customer-id']
+        );
+    }
+
+    /**
+     * Fetch recommendations from YooChoose
+     *
+     * @param $userId
+     * @param $scenarioId
+     * @param $locationId
+     * @param $limit
+     * @return json|null
+     */
+    public function getRecommendations( $userId, $scenarioId, $locationId, $limit )
+    {
+        $extension = "json";
+
+        $uri = $this->getRecommendationEndpoint();
+        $uri .= "/$userId/$scenarioId.$extension?numrecs=$limit";
```

Comment:
For me I makes sense to check if those parameters are set. If not throw `InvalidArgumentException`.